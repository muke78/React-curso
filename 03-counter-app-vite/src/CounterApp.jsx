import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  const handleAdd = () => {
    setCount(count + 1);
    // setCount((c) => c + 1);
  };

  const handleDelete = () => {
    setCount(count - 1);
  };

  const handleReset = () => {
    setCount(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{count}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleDelete}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
