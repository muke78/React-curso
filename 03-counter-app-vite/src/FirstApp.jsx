// Manera normal de encapsulamiento de componentes sin ocupar div

import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <code> {JSON.stringify(newMessage)} </code> */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};
// PropTypes
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  name: PropTypes.string,
};

/**
 * Las default props entras primero que las propTypes 
 * (compatibilidad con defaultProps se eliminará de los componentes 
 * de funciones en una versión importante futura.) */

// FirstApp.defaultProps = {
//   name: "Erick Gonzalez",
//   subtitle: "No hay subtitulo",
//   // title: "No hay ningun titulo",
// };

