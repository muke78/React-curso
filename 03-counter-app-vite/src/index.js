export * from './CounterApp';
export * from './FirstApp';
export * from './HelloWorldApp';
export * from './main';
