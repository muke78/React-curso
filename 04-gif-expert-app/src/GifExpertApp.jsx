export const GifExpertApp = () => {
  return (
    <>
      {/* titulo */}
      <h1>GiftExpertApp</h1>

      {/* input */}

      {/* Listado de Gif  */}
      {/* Gif Item  */}
    </>
  );
};
