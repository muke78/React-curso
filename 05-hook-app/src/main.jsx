import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook';
// import { CounterApp } from './01-useState/CounterApp';
// import { HooksApp } from './HooksApp';
// import { SimpleForm } from './02-useEffect/SimpleForm';
import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';

import './index.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <FormWithCustomHook />
  </StrictMode>,
)
