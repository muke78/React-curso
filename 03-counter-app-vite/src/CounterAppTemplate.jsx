import { CounterApp } from "./CounterApp";
export const CounterAppTemplate = () => {
  return <CounterApp value={ 0 } />;
};
