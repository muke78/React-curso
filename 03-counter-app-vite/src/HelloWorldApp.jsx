import { CounterApp } from "./CounterApp";
export const HelloWorldApp = () => {
  return <CounterApp value={ 0 } />;
};
