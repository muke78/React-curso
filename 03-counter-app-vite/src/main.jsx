import React from "react";
import ReactDOM from "react-dom/client";
import { CounterAppTemplate } from "./CounterAppTemplate";
import { FirstApp } from "./FirstApp";
import "./style.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <CounterAppTemplate />
    {/* <FirstApp
      title="Hola, soy Erick"
      subtitle="Hola soy un subtitulo"
    /> */}
  </React.StrictMode>
);
