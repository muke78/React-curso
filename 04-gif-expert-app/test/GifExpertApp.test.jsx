import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Pruebas en el <GifExpertApp/>', () => {
    test('debe de cambiar mi value en el input ', () => {
        render(<GifExpertApp />)
        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: 'Demon Slayer' } });
        expect(input.value).toBe('Demon Slayer')
    });

    test('no mostrar una categoria si ya existe', () => {

        render(<GifExpertApp />)
        const inputValue = 'Demon slayer';
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue } });
        fireEvent.submit(form);

        expect(screen.getAllByText('Demon Slayer').length).toBe(1);
    });

    test('Hacer un snapshot del componente', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });
})