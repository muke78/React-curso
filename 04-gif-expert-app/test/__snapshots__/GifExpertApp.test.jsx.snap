// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifExpertApp/> Hacer un snapshot del componente 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifts"
      type="text"
      value=""
    />
  </form>
  <h3>
    Demon Slayer
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
